### ISSUE TITLE
"Volumn for domain is already created." is shown over and over again after error occurs during `vagrant up'

### ISSUE THREAD

**Issue by tkdchen** [vagrant-libvirt/vagrant-libvirt#447]
Version `0.0.26`

Steps to reproduce:
1. `vagrant up --provider=libvirt`
2. some error occurs and process is interrupted
3. correct the error
4. rerun `vagrant up --provider=libvirt`, following message comes out
   
   ```
   Volume for domain is already created. Please run 'vagrant destroy' first.
   ```
5. run `vagrant destroy`
6. then repeat step 4, same message comes out again
7. repeat 5 and 6, I'm asked to run `vagrant destroy` over and over again

**Comment by dsager** [comment]
I am having the same issue, did you find a way to make it work?

**Comment by dsager** [comment]
i found a workaround:
adding this to my `Vagrantfile` solved the issue:
```ruby
config.vm.provider :libvirt do |domain, override|
  # ...
  domain.random_hostname = true
end
```

**Comment by electrofelix** [comment]
This seems like some state got lost, which can happen if clean up is incomplete.

@tkdchen have you seen this occur with a more recent version?

I'm wondering what would be a reasonable solution? Check to see if any of the volumes exist before launching and report them all and then abort with a message to tell the user that if they are stray volumes to delete them by hand?

**Comment by electrofelix** [comment]
@tkdchen that's understandable, I don't think we need to reproduce, just wondering what you think might have been a reasonable outcome if given that it's probably a bad idea to delete a volume that the code can't be 100% sure belongs to the environment the user is working with. 

Would it be sufficient to update the error message to suggest to the user how to manually remove if they believe the volumes should not exist?

I can see if it simply creating a volume with the same name causes the issue and if there is no machine associated consider asking the user if it's ok to delete the volume on destroy (or delete it automatically if the force option is set), but not sure if that will be feasible.

Just looking to see if these would have been reasonable outcomes from your perspective, even if you can't test it any more.

**Comment by electrofelix** [comment]
I believe this was fixed shortly after this issue was logged by the addition of a remove stale volume action that was merged in #377 on the 23rd May 2015 and release in 0.0.27. Sorry hadn't spotted that the last time I looked at this.

Based on that, going to close this.
